### PR TITLE
Use `psa` for compilation in Nix derivations

### LINKED ISSUE #518: Use `psa` for compilation
This way we can censor warnings from dependencies during compilation

### PR DESCRIPTION
**Pull request by ngua**
Closes #518. Now we can compile with `psa --strict --censor-lib` which will silence the superfluous warnings from libraries and cause building to fail if any compiler warnings are reported (very helpful to catch things like redundant imports, etc...)

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [-] All Purescript imports are explicit, including constructors (n/a)
- [-] **All** existing examples have been run locally against a fully-synced testnet node (n/a)
- [-] The integration and unit tests have been run (`npm run test`) locally (n/a, except I ran the `checks.ctl-unit-test` via Nix to make sure it passes (Hydra is having issues)

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -164,6 +142,73 @@ let
...
+      });
+
+  buildPursDocsSearch =
+    { name ? "purescript-docs-search"
+    , ...
+    }:
+    let
+      docsUrl =
+        "https://github.com/purescript/purescript-docs-search/releases/download";
+      docsVersion = "v0.0.11";
```
Comment (bladyjoker) [inline nix/default.nix:185]: Consider using flake inputs for this and then `dependency inject` this like:

```
    purs-docs-search-app = {
      url  = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/docs-search-app.js";
      flake = false;
      };
    purs-docs-search = {
      url  = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/purescript-docs-search";
      flake = false;
      };
```
But I understand the convenience of the current approach, so which ever you find nicer...

File: nix/default.nix [hunk 2]
```diff
@@ -134,7 +111,8 @@ let
           install-spago-style
         '';
       buildPhase = ''
-        build-spago-style "./**/*.purs"
+        psa --strict --censor-lib --is-lib=.spago ${spagoGlobs} \
+          --censor-codes=UserDefinedWarning "./**/*.purs"
```
Comment (bladyjoker) [inline nix/default.nix:115]: This is great! I wish psa had config file to manage reporting like hlint does. Consider adding `censorCodes` argument with `[ "UserDefinedWarning" ]` as default

**Review by bladyjoker — APPROVED** [review]
Added some comments, mostly nice-to-haves. Didn't know PSA was a thing, very useful!

File: nix/default.nix (hunk 1 above)
Comment (ngua) [reply nix/default.nix:185]: Yes, that would be ideal

**Pull request merged**